waspc/src/Wasp/Generator/FileDraft/CopyDirFileDraft.hs

```diff
@@ -35,6 +42,8 @@ data CopyDirFileDraft = CopyDirFileDraft
 instance Writeable CopyDirFileDraft where
   write projectRootAbsPath draft = do
     srcDirExists <- doesDirectoryExist $ SP.fromAbsDir srcPathAbsDir
+    dstDirExists <- doesDirectoryExist $ SP.fromAbsDir dstPathAbsDir
+    when dstDirExists $ removeDirectoryRecursive dstPathAbsDir
```

Comment:
I added this for two reason:
1) It now matches the semantics of copy file, where if it exists it overwrites it and thus they are equal. Now, when we have a dir FD, we know they will be equal (i.e. the destination cannot have "extra" stuff).
2) When a user wants to remove a migration from their wasp project dir, we want that to be removed from the generated project. This may happen when they do a `--create-only` but then don't migrate and just remove the dir. Without this, it would stay in their generated project and the next migrate could apply it.

Reply:
I don't think this will negatively affect the generation process. The only thing it should do is ensure there is nothing stale leftover, which is what we want. I don't think we have `createDirectory` alone sadly. Prob ok to just use what we have for now I think.